src/values/syntax.rs

```diff
@@ -200,7 +200,14 @@ impl<'i> SyntaxString {
   ) -> Result<ParsedComponent<'i>, ParseError<'i, ParserError<'i>>> {
     match self {
       SyntaxString::Universal => Ok(ParsedComponent::Token(crate::properties::custom::Token::from(
-        input.next()?,
+        match input.next_including_whitespace() {
+          Ok(token) => token,
+          Err(BasicParseError {
+            kind: BasicParseErrorKind::EndOfInput,
+            ..
+          }) => &Token::WhiteSpace(""),
+          Err(e) => return Err(e.into()),
+        },
```

Comment:
I guess as you mentioned we might want this function to require a value in other cases, e.g. when parsing custom at rules (which also use this grammar). One idea would be to move this to `property.rs` when parsing the `initial-value` property. There is already some special handling for universal syntax strings there.

https://github.com/parcel-bundler/lightningcss/blob/b543fe7fc96239d35b55d1d9ff5cf1f3dd170a8a/src/rules/property.rs#L78-L85

so basically something like 

```rust
match syntax.parse_value(&mut parser) {
  Ok(v) => Some(v),
  Err(BasicParseError {
     kind: BasicParseErrorKind::EndOfInput,
      ..
  }) => ParsedComponent::Token(Token::WhiteSpace("")),
  Err(e) => return Err(e)
}

Reply:
or maybe better than checking for an error, you could use `parser.is_exhausted()` to tell if it's already at the end

Reply:
Yep, the only problem I see with `parser.is_exchausted()` is that it will be true for both `initial-value:;` and `initial-value: ;` and in the second case we _do_ want to know the space exists or not to be collected.

The nice part I think is that `is_exhausted()` would be files if an actual error apart from EndOfInput occured. Therefore we can match all `Err` safely, I think.

I've updated the code with your suggestions and with the above as well. I think this looks better than the original version I implemented, does this look good to you too now that it is hoisted?

Reply:
Wait hold on. I think both `:;` and `: ;` should be allowed, but maybe they both end up as `: ;`. That makes it even simpler.

Let me update the code for that.

Reply:
Alright updated. Right now both `initial-value:;` and `initial-value: ;` and even `initial-value:         ;` should be parseable. But in all these scenarios we now print it as `initial-value: ;`.

If you don't agree with this change, we should be able to revert the last commit 👍